library/database/class.sqldriver.php

```diff
@@ -1567,9 +1567,7 @@ public function orWhereNotIn($field, $values) {
      * @return string The parsed expression.
      */
     protected function _parseExpr($expr, $name = null, $escapeExpr = false) {
-        $result = '';
-
-        $c = substr($expr, 0, 1);
```

Comment:
This was causing a warning when `$expr` is not a string. We had made a change a while back to accept dates so it's a real possibility.